### PR TITLE
Default to regional STS endpoints (V2)

### PR DESCRIPTION
**Pull request by alextwoods**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

1. To make sure we include your contribution in the release notes, please make sure to add description entry for your changes in the "unreleased changes" section of the `CHANGELOG.md` file (at corresponding gem). For the description entry, please make sure it lives in one line and starts with `Feature` or `Issue` in the correct format.

2. For generated code changes, please checkout below instructions first:
  https://github.com/aws/aws-sdk-ruby/blob/master/CONTRIBUTING.md

Thank you for your contribution!

### CODE REVIEW

**Comment by mullermp** [comment]
See comments on V3.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Unreleased Changes
 ------------------
 
-* Feature - Default to regional STS endpoints
+* Issue - Change the default of sts_regional_endpoints from 'legacy' to 'regional'.
```
Comment (mullermp) [inline CHANGELOG.md]: Use markdown `sts_regional_endpoints`  :)

File: aws-sdk-core/spec/aws/sts/sts_regional_endpoints_spec.rb [hunk 2]
```diff
@@ -27,15 +27,15 @@ module STS
           expect(client.config.sts_regional_endpoints).to eq('regional')
         end
 
-        it 'defaults to `legacy`' do
+        it 'defaults to `regional`' do
```
Comment (mullermp) [inline aws-sdk-core/spec/aws/sts/sts_regional_endpoints_spec.rb]: These backticks should be removed too yeah?

Comment (alextwoods) [reply aws-sdk-core/spec/aws/sts/sts_regional_endpoints_spec.rb]: Yeah agreed.  These won't execute `regional` but I think better style would be: "defaults to 'regional'"

**Pull request merged**